src/runtime/pay-client.js

```diff
@@ -65,11 +65,14 @@ export class PayClient {
    * @param {!Window} win
    * @param {!../components/activities.ActivityPorts} activityPorts
    * @param {!../components/dialog-manager.DialogManager} dialogManager
+   * @param {!../runtime/analytics-service.AnalyticsService} analyticsService
    */
-  constructor(win, activityPorts, dialogManager) {
+  constructor(win, activityPorts, dialogManager, analyticsService) {
```

Comment:
Will it ever need the entire analytics service or can we just pass the transaction ID?  Might make some of your tests simpler with just the ID.

Reply:
I need this analyticsService here to generate a new ID.

Reply:
Where is it generating a new ID after the pay client gets instantiated?  I don't see that anywhere.  I wouldn't think it should ever do that.

Also, just considering possibilities but should PayClient take a deps object instead of all these individual parameters?

Reply:
Line 71 generates a new ID.

Reply:
I think code may have shifted around.  Are you talking about: analyticsService.getTransactionId()?

Reply:
analyticsService.getTransactionId() was used in the original code in pay-flow.js to assign a transaction ID to i param in payment request. I moved it to pay-client.js to use it before PaymentsAsyncClient is initialized. Happy to talk offline.

Reply:
Comment has been addressed in the new commit.